src/main/java/net/dv8tion/jda/internal/entities/detached/DetachedRoleImpl.java

```diff
@@ -0,0 +1,346 @@
+    @Override
+    public EnumSet<Permission> getPermissions(@Nonnull GuildChannel channel)
+    {
+        throw detachedException();
+    }
+
+    @Nonnull
+    @Override
+    public EnumSet<Permission> getPermissionsExplicit()
+    {
+        return getPermissions();
+    }
+
+    @Nonnull
+    @Override
+    public EnumSet<Permission> getPermissionsExplicit(@Nonnull GuildChannel channel)
+    {
+        throw detachedException();
+    }
+
+    @Override
+    public Color getColor()
+    {
+        return color != Role.DEFAULT_COLOR_RAW ? new Color(color) : null;
+    }
+
+    @Override
+    public int getColorRaw()
+    {
+        return color;
+    }
+
+    @Override
+    public boolean isPublicRole()
+    {
+        return getIdLong() == this.getGuild().getIdLong();
+    }
+
+    @Override
+    public boolean hasPermission(@Nonnull Permission... permissions)
+    {
+        long effectivePerms = rawPermissions;
+        for (Permission perm : permissions)
+        {
+            final long rawValue = perm.getRawValue();
+            if ((effectivePerms & rawValue) != rawValue)
+                return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean hasPermission(@Nonnull GuildChannel channel, @Nonnull Permission... permissions)
+    {
+        throw detachedException();
+    }
+
+    @Override
+    public boolean canSync(@Nonnull IPermissionContainer targetChannel, @Nonnull IPermissionContainer syncSource)
+    {
+        throw detachedException();
+    }
+
+    @Override
+    public boolean canSync(@Nonnull IPermissionContainer channel)
+    {
+        throw detachedException();
+    }
+
+    @Override
+    public boolean canInteract(@Nonnull Role role)
+    {
+        // Works, depends on compareTo
+        return PermissionUtil.canInteract(this, role);
+    }
+
+    @Nonnull
+    @Override
+    public Guild getGuild()
+    {
+        return guild;
+    }
+
+    @Nonnull
+    @Override
+    public RoleManager getManager()
+    {
+        throw detachedException();
+    }
+
+    @Nonnull
+    @Override
+    public AuditableRestAction<Void> delete()
+    {
+        throw detachedException();
+    }
+
+    @Nonnull
+    @Override
+    public JDA getJDA()
+    {
+        return api;
+    }
+
+    @Nonnull
+    @Override
+    public RoleTags getTags()
+    {
+        return tags == null ? RoleTagsImpl.EMPTY : tags;
+    }
+
+    @Nullable
+    @Override
+    public RoleIcon getIcon()
+    {
+        return icon;
+    }
+
+    @Nonnull
+    @Override
+    public String getAsMention()
+    {
+        return isPublicRole() ? "@everyone" : "<@&" + getId() + '>';
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return id;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (o == this)
+            return true;
+        if (!(o instanceof Role))
+            return false;
+        Role oRole = (Role) o;
+        return this.getIdLong() == oRole.getIdLong();
+    }
```

Comment:
This would imply a detached role and an not-detached one are equal, is this intended?